build/patches/Enable-crash-reporting-ui.patch

```diff
@@ -0,0 +1,206 @@
+-                Log.i(TAG, "Minidump cannot currently be uploaded due to network constraints.");
+-                return MinidumpUploadStatus.FAILURE;
+-            }
+-        }
+-
+         MinidumpUploader.Result result = mMinidumpUploader.upload(mFileToUpload);
+         if (result.isSuccess()) {
+             String uploadId = result.message();
+diff --git a/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/MinidumpUploader.java b/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/MinidumpUploader.java
+--- a/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/MinidumpUploader.java
++++ b/components/minidump_uploader/android/java/src/org/chromium/components/minidump_uploader/MinidumpUploader.java
+@@ -11,6 +11,8 @@ import java.io.BufferedReader;
+ import java.io.ByteArrayOutputStream;
+ import java.io.File;
+ import java.io.FileInputStream;
++import java.io.FileOutputStream;
++import java.io.BufferedOutputStream;
+ import java.io.FileReader;
+ import java.io.IOException;
+ import java.io.InputStream;
+@@ -18,6 +20,8 @@ import java.io.OutputStream;
+ import java.net.HttpURLConnection;
+ import java.util.zip.GZIPOutputStream;
+ 
++import org.chromium.base.PathUtils;
++
+ /**
+  * This class tries to upload a minidump to the crash server.
+  *
+@@ -120,30 +124,18 @@ public class MinidumpUploader {
+             if (fileToUpload == null || !fileToUpload.exists()) {
+                 return Result.failure("Crash report does not exist");
+             }
+-            HttpURLConnection connection =
+-                    mHttpURLConnectionFactory.createHttpURLConnection(CRASH_URL_STRING);
+-            if (connection == null) {
+-                return Result.failure("Failed to create connection");
+-            }
+-            configureConnectionForHttpPost(connection, readBoundary(fileToUpload));
+-
+-            try (InputStream minidumpInputStream = new FileInputStream(fileToUpload);
+-                    OutputStream requestBodyStream =
+-                            new GZIPOutputStream(connection.getOutputStream())) {
+-                streamCopy(minidumpInputStream, requestBodyStream);
+-                int responseCode = connection.getResponseCode();
+-                if (isSuccessful(responseCode)) {
+-                    // The crash server returns the crash ID in the response body.
+-                    String responseContent = getResponseContentAsString(connection);
+-                    String uploadId = responseContent != null ? responseContent : "unknown";
+-                    return Result.success(uploadId);
+-                } else {
+-                    // Return the remote error code and message.
+-                    return Result.uploadError(responseCode, connection.getResponseMessage());
+-                }
+-            } finally {
+-                connection.disconnect();
+-            }
++
```

Comment:
What is `fileToUpload`? The zip file, the content of the zip file or the `upload_file_minidump` alone?

Reply:
The zip file, exactly this https://github.com/bromite/bromite/issues/944#issuecomment-825598966

Reply:
Ok, then it is not necessary to compress it again? Since it is already a zip file.

Reply:
> Ok, then it is not necessary to compress it again?

sorry I didn't understand, no it's me with that code that creates the zip

Reply:
Then you should call the file `.gz`, it is not a `.zip`.

Reply:
Would it be possible to avoid the MIME container and use `ZipOutputStream` directly instead? Then we don't mess with MIME and gzip. It is not a problem for developers but .zip would allow users to inspect more easily what they are submitting.

Reply:
> Would it be possible to avoid the MIME container

I don't think it's easily possible.
for the zip no problem

Reply:
I went to check how the `upload()` function is called:

1. what happens when multiple local minidumps are available?
2. are the internal minidump files deleted after user saves them?

Reply:
> what happens when multiple local minidumps are available?

a picture is more than a thousand words
![image](https://user-images.githubusercontent.com/29201891/116212093-c46ca000-a744-11eb-8d69-92c99d276226.png)

> are the internal minidump files deleted after user saves them?

no, they stay there
there is a whole self-cleaning system already active (even today) that deletes the files sent and not after a number of days

Reply:
added `ZipOutputStream` in https://github.com/bromite/bromite/pull/1105/commits/8fda0c2fb16dfc9b5381775596b8bd7191fb54b0